File: integrations/src/main/java/com/nivasafinance/services/voice/dto/VoiceCallRequest.java
package com.nivasafinance.services.voice.dto;

import com.nivasafinance.common.enums.SystemEntities;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class VoiceCallRequest {
    private String fromNumber;
    private String toNumber;
    private String callerId;
    private CallBackData callBackData;

    @AllArgsConstructor
    @NoArgsConstructor
    @Data
    public static class CallBackData{
        private SystemEntities entityType;
        private String identifier;
    }
}

File: integrations/src/main/kotlin/services/voice/dto/VoiceCallRequest.kt
package services.voice.dto

data class VoiceCallRequest(
    val fromNumber: String,
    val toNumber: String,
    val exophone: String? = null,
    val entityName: String? = null,
    val entityId: Long? = null,
    val callPurpose: String? = null
)
